src/backend/utils/adt/agtype.c

```diff
@@ -2704,10 +2774,14 @@ Datum agtype_to_int8(PG_FUNCTION_ARGS)
         if (temp->type != AGTV_ARRAY ||
             !temp->val.array.raw_scalar)
         {
+            int type = temp->type;
+
+            pfree_agtype_value(temp);
```

Comment:
Shouldn't agtv be freed here as well?

Reply:
`ereport(ERROR` frees memory, so cleanup isn't necessary here. Additionally, I have removed all of the cleanups preceding `ereport(ERROR`